sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -884,7 +900,12 @@ private void validateReplicaAddresses(String collectionRid, AddressInformation[]
 
         if (addressesNeedToValidation.size() > 0) {
             this.openConnectionsHandler
-                    .openConnections(collectionRid, this.serviceEndpoint, addressesNeedToValidation)
+                    .openConnections(
```

Comment:
should not all the open connections go throughput OpenConnectionsProcessor? instead some goes through openConnectionProcessor vs openConnectionHandler? 

And also currently openConnectionHandler does not have any concurrency checking, so by calling it directly will have a risk of opening connection too aggressive?